server/interpreter.js

```diff
@@ -3096,17 +3091,17 @@ Interpreter.prototype['stepSwitchStatement'] = function(stack, state, node) {
     if (switchCase) {
       if (!state.matched_ && !stack.tested_ && switchCase['test']) {
         stack.tested_ = true;
-        this.pushNode_(switchCase['test']);
-        return;
+        return new Interpreter.State(switchCase['test'], state.scope);
       }
       if (state.matched_ || state.value === state.switchValue_) {
         state.matched_ = true;
         var n = state.n_ || 0;
         if (switchCase['consequent'][n]) {
           state.isSwitch = true;
-          this.pushNode_(switchCase['consequent'][n]);
+          var nextState = new Interpreter.State(
+              switchCase['consequent'][n], state.scope);
           state.n_ = n + 1;
```

Comment:
Not quite: the line is 81 characters.  Guess I could omit the final semicolon, seeing as it's the last statement in the block?  :-)